nni/compression/pytorch/speedup/compress_modules.py

```diff
@@ -152,24 +163,120 @@ def replace_conv2d(conv, mask):
         for groupid in range(conv.groups):
             start = groupid * input_step
             end = (groupid + 1) * input_step
-            current_input_index = list(filter(lambda x: start <= x and x < end, in_channels_index.tolist()))
+            current_input_index = list(
+                filter(lambda x: start <= x and x < end, in_channels_index.tolist()))
             if not current_input_index:
                 # there is no kept channel in current group
-                continue
+                # TODO bug here, the groups is directly get from conv.groups, if the whole group is removed,
+                # then the number of groups in the new_conv also need to change
+                raise Exception(
```

Comment:
do we support depthwise conv?

Reply:
Yes, we do. We can only support pruning the whole group of filters for the depthwise conv.